src/Tribe/Rewrite.php

```diff
@@ -589,6 +591,11 @@ public function get_canonical_url( $url, $force = false ) {
 			$this->canonical_url_cache[ $url ] = $resolved;
 		}
 
+		if ( count( $passthru_vars ) ) {
+			// If there are pass-through query vars, then re-apply them to the clean URL after it's been cached.
+			$resolved = add_query_arg( $passthru_vars, $resolved );
```

Comment:
By doing this here - won't it cache the URL with no passthrough vars, then later return the cached URL w/o them since it's cached w/o them?

Reply:
Excellent point, I've addressed it in my latest commit and improved the code a bit, thanks.